frontend/src/components/dashboard/create/email/EmailTemplate.tsx

```diff
@@ -30,29 +35,59 @@ const EmailTemplate = ({
   const bodyPlaceholder =
     'Dear {{ name }}, your next appointment at {{ clinic }} is on {{ date }} at {{ time }}'
 
-  async function handleSaveTemplate(): Promise<void> {
-    setErrorMsg(null)
-    try {
-      if (!campaignId) {
-        throw new Error('Invalid campaign id')
+  const handleSaveTemplate = useCallback(
+    async (next = true): Promise<void> => {
+      setErrorMsg(null)
+      try {
+        if (!campaignId) {
+          throw new Error('Invalid campaign id')
+        }
+        const { updatedTemplate, numRecipients } = await saveTemplate(
+          +campaignId,
+          subject,
+          body,
+          replyTo
+        )
+        onNext(
+          {
+            subject: updatedTemplate?.subject,
+            body: updatedTemplate?.body,
+            replyTo: updatedTemplate?.reply_to,
+            params: updatedTemplate?.params,
+            numRecipients,
+          },
+          next
+        )
+      } catch (err) {
+        setErrorMsg(err.message)
+
+        // Rethrow error to be handled elsewhere
+        if (!next) {
+          throw err
+        }
       }
-      const { updatedTemplate, numRecipients } = await saveTemplate(
-        +campaignId,
-        subject,
-        body,
-        replyTo
+    },
+    [body, campaignId, onNext, replyTo, subject]
+  )
+
+  // Set callback for finish later button
+  useEffect(() => {
+    finishLaterCallbackRef.current = () => {
+      modalContext.setModalContent(
+        <SaveDraftModal
+          saveable
+          onSave={async () => {
+            if (subject && body) {
+              await handleSaveTemplate(false)
```

Comment:
On second look, I think the original intention of `next = false` was so that the `onNext` callback will not advance to the next step in the `SMSCreate` component (as an example). 

But this has no effect. After the draft is saved, any time the user selects the same campaign again from the dashboard, it will be automatically advanced to Step 2. This is due to the fact that the progress is derived based on the presence of a template body on the campaign fetched from the API. 

The only use for this argument now is just a flag to indicate whether to re-throw the error. Perhaps I can rename this argument?

Reply:
Yea I think ideally if 'Finish this later' is clicked we should retain at the step, but I believe that will require a much bigger change (ie changing the logic of how progress is derived).

So i guess yup for now maybe we can rename this argument?